### DISCUSSION TITLE
Dag to do backfills of other dags

### DISCUSSION THREAD

**Discussion by match-gabeflores** [apache/airflow#28390] [Q&A]
hi, since airflow isn't able to do backfills in the UI (https://github.com/apache/airflow/issues/14854), we are thinking of just creating a simple dag that calls the airflow `CLI ` to do the backfill. the idea is we just trigger this "`backfill caller`" dag with the dag_id (and parameters) that needs backfill.

Does anyone see issues with this or done something similar?

**Comment by potiuk** [comment]
Some people do it and it worked for them. As long you are able to handle failures manually, this is no different than automating any other tasks with Airflow. This is equivalent of running a "bash" operator with potentially long running operation. As long as the DAG is different , than the dag it backfills, there are no risks for contentions, so no red flags there.

**Reply by match-gabeflores** [reply]
Thanks for the sanity check. That aligns with our thourghts. We will work on something like this and report back any challenges.

**Comment by bhulsey** [comment]
Any progress here? I am a newbie and have a scenario that requires backfill for non-consecutive data intervals. 

My Dag runs each minute, waiting for a specific set of files, then logs them to a database. Occasionally files show up very late or not at all. I need to scan the database daily to backfill (clear & rerun) any times for which files were missed.  

The only working solution I've come up with so far is "brute force" looping over the backfill command using BashOperator or subprocess. (I can also loop over TriggerDagRunOperator but I'm having trouble aligning 'execution_date' with 'data_interval_start'.)

If I have a list of 'data_interval_start' times, is there any other way to automatically rerun those dag runs?

**Comment by Fabhi** [comment]
I tried implementing a "backfiller" dag to do this, and many a times it would trigger one dagrun out of the 10 requested ones but with the type as "scheduled" rather than "backfill" and blocks itself from running all other runs with the error " Backfill cannot be created for DagRun scheduled__2025-10-20T02:00:00+00:00 in 2025-10-20T02:00:00, as there's already scheduled in a RUNNING state"

Did you guys also encounter this issue? @bhulsey @match-gabeflores

**Reply by bhulsey** [reply]
@Fabhi Instead of tracking Airflow failures, we've added a custom Postgres backend to log our incoming data and output. We use this for tracking now and simply trigger a new dag run as needed for failures or reprocessing. If you haven't upgraded to version 3, perhaps the new API has some improvements you could use.
